Ct 3922 create sar ir case

willmcb opened the pull request:
## Description
Adds page to enter details of the SAR IR with relevant details pulled from the original SAR.

Note I am going to rails another PR with page objects for the feature tests. This PR was getting big so I wanted to get it reviewed before I made further changes.

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [x] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed


### Screenshots
Adds new page where user can enter the SAR IR case details:

<img width="333" alt="Screenshot 2021-11-30 at 15 58 52" src="https://user-images.githubusercontent.com/13377553/144082146-0cd401ec-3431-4f19-af25-287d5a04dbd3.png">

### Related JIRA tickets
[CT-3922](https://dsdmoj.atlassian.net/browse/CT-3922)
[CT-3923](https://dsdmoj.atlassian.net/browse/CT-3923)

### Manual testing instructions
- copy and existing SAR case number
- click create case
- enter copied SAR number
- Continue
- Add the extra details to the page
- Continue - should be directed to the case show page (not the case details partial for this is a placeholder at the moment)

Commit:
Add method to .humanise subject type

Commit:
Add test for page rendering with add case header

Commit:
Add small case details field at top of page + test

Commit:
Basic form with partial

Commit:
Validations working as expected

Commit:
complete basic create spec

Commit:
Test for created case

Commit:
Add sar_ir_subtypes

Commit:
Display Original case details on SAR IR subject

Commit:
Make carried over attributes more testable

Commit:
Change dropzone copy

Commit:
Add default field to make SAR IRs triggered cases

Also fixes test failure issue

digitalali-moj reviewed (commented):
Nice f/e work 👍 
Minor front end questions / potential changes.

ymao2 reviewed (commented):
I have some comments and some confuses, happy to discuss them

Commit:
Make test more resiliant

Commit:
Remove superflous validation declarations

Commit:
Fix error messages

Commit:
Fix spec

Commit:
Remove call to validate steps if case is final step

This avoids having to add a lot of validation functions in the parent
'Standard' class as step validations are really simple (just link /
confirm step now) and not contingent on conditional logic. Conditional
logic was not being called by .attributes_valid? in ApplicationRecord
which meant having to declare loads of validation methods in Standard
SAR class.

This rework gets around this by not calling the step validations on the
final step and instead just validating the case completely.

Commit:
Add text to en.yml or rely on defaults from InternalReview case

Commit:
Use hidden inputs for default fields

Commit:
Add individual page titles for steps

Commit:
Fix spec

ymao2 reviewed (approved):
Nice work and also the changes you made for the validation parts ! 👍

digitalali-moj reviewed (approved):
Brilliant work! couple of minor comments but easy fixes.
